v8go.cc

```diff
@@ -600,16 +606,22 @@ ContextPtr NewContext(IsolatePtr iso,
   return ctx;
 }
 
+int ContextRetainedValueCount(ContextPtr ctx) {
+  return ctx->vals.size();
+}
+
 void ContextFree(ContextPtr ctx) {
   if (ctx == nullptr) {
     return;
   }
   ctx->ptr.Reset();
 
-  for (m_value* val : ctx->vals) {
-    val->ptr.Reset();
-    delete val;
+  
+  for(auto& [key, value] : ctx->vals) {
+    value->ptr.Reset();
+    delete value;
   }
+  ctx->vals.clear();
```

Comment:
:no_bell: I don't know about perf but I think it'd be clearer to call ValueRelease() (but be careful about modifying while looping).